core/arch/aarchxx/mangle.c

```diff
@@ -777,7 +777,9 @@ insert_parameter_preparation(dcontext_t *dcontext, instrlist_t *ilist, instr_t *
     for (i = 0; i < num_regs; i++) {
         if (opnd_is_immed_int(args[i]))
             regs[i] = -1;
-        else {
+        else if (opnd_is_base_disp(args[i])) {
+            regs[i] = -3;
```

Comment:
What does -3 mean? Comment block above goes up to -2.

Reply:
-3 is opnd_is_base_disp type (previously unsupported) as described in the condition

Reply:
Will add to the comment

Reply:
I think it would be better if these were enums or at least macro defs to aid readability.